### PR TITLE
[mono][jit] ArmBase methods are now intrinsic.

### PR DESCRIPTION
**Pull request by jandupej**
This adds `ArmBase` methods as intrinsic under arm64. These see use in BCL and might result in minor performance improvements.

EDIT: The support for `ArmBase.Yield` is now also added on both LLVM and mini sides.

### CODE REVIEW

File: src/mono/mono/mini/simd-intrinsics.c [hunk 1]
```diff
@@ -3596,6 +3597,13 @@ emit_arm64_intrinsics (
 				(is_64bit ? OP_XOP_I8_I8 : OP_XOP_I4_I4),
 				(is_64bit ? INTRINS_BITREVERSE_I64 : INTRINS_BITREVERSE_I32),
 				arg0_type, fsig, args);
+		case SN_Yield: {
+			MonoInst* ins;
+			MONO_INST_NEW (cfg, ins, OP_NOP);
```
Comment (vargaz) [inline src/mono/mono/mini/simd-intrinsics.c:3602]: arm64 has a yield instruction.

**Review by vargaz — APPROVED** [review]

File: src/mono/mono/mini/simd-intrinsics.c (hunk 1 above)
Comment (jandupej) [reply src/mono/mono/mini/simd-intrinsics.c]: Added `OP_ARM64_HINT` to account for this and other hint codes. This is also reflected on LLVM side.

**Comment by jandupej** [comment]
CI failures are unrelated. Will merge on Tue.

**Pull request merged**